jax/_src/scipy/sparse/linalg.py

```diff
@@ -277,15 +273,36 @@ def cg(A, b, x0=None, *, tol=1e-5, atol=0.0, maxiter=None, M=None):
       inverse of A.  Effective preconditioning dramatically improves the
       rate of convergence, which implies that fewer iterations are needed
       to reach a given error tolerance.
+  assume_ipart_is_zero : bool, optional
+      Whether the linear operator dtype can be assumed to have imag(A)=0.
+      Defaults to False for complex-value systems.
```

Comment:
We should add a "why" to this: something like "when True, then it can be assumed that for complex operators `A`, `A @ x` and `A.T @ x` are equivalent, leading to more efficient reverse-mode autodiff."